src/DistributedComputations/distributed_architectures.jl

```diff
@@ -230,11 +231,22 @@ function Distributed(child_architecture = CPU();
     # The rank connectivity _ALWAYS_ wraps around (The cartesian processor "grid" is `Periodic`)
     local_connectivity = RankConnectivity(local_index, ranks) 
 
-    # Assign CUDA device if on GPUs
+
+    # Assign GPUs to each MPI rank
     if child_architecture isa GPU
         local_comm = MPI.Comm_split_type(communicator, MPI.COMM_TYPE_SHARED, local_rank)
         node_rank  = MPI.Comm_rank(local_comm)
-        isnothing(devices) ? device!(node_rank % ndevices()) : device!(devices[node_rank+1]) 
+        if child_architecture == CUDAGPU()
+            device_id = isnothing(devices) ? node_rank % CUDA.ndevices() : devices[node_rank+1]
+            CUDA.device!(device_id)
+        end
+        if child_architecture == ROCmGPU()
+            device_id = isnothing(devices) ? node_rank % length(AMDGPU.devices()) : devices[node_rank+1]
+            AMDGPU.device!(device_id)
+
+            # gpu_device = isnothing(devices) ? AMDGPU.devices(:gpu)[node_rank % length(AMDGPU.devices(:gpu))+1] : AMDGPU.devices(:gpu)[devices[node_rank+1]]
+            # AMDGPU.default_device!(gpu_device)
```

Comment:
why commented code?

Reply:
Because I forgot to remove it.. Thanks for catching it. Fixed.